uni/lib/controller/local_storage/preferences_controller.dart

```diff
@@ -71,17 +72,18 @@ class PreferencesController {
   }
 
   /// Saves the user's student number, password and faculties.
+  static const FlutterSecureStorage storage = FlutterSecureStorage();
   static Future<void> savePersistentUserInfo(
     String user,
     String pass,
     List<String> faculties,
   ) async {
-    await prefs.setString(_userNumber, user);
-    await prefs.setString(_userPw, encode(pass));
-    await prefs.setStringList(
-      _userFaculties,
-      faculties,
-    ); // Could be multiple faculties
+    await storage.write(key: _userNumber, value: user);
+    await storage.write(key: _userPw, value: pass);
+    await storage.write(
```

Comment:
You don't need to store the faculties privately. Also, doing this without changing the faculties getter is probably not right